core/libs/load-env-vars.el

```diff
@@ -87,12 +87,10 @@
       (let ((key (car element)) (value (cadr element)))
         (if (string-equal "PATH" key)
             (let ((paths (split-string value path-separator)))
-              (dolist (p paths)
-                (add-to-list 'exec-path (directory-file-name
-                                         (subst-char-in-string ?\\ ?/ p))
-                             'append))
-              (setenv "PATH" (mapconcat convert-to-os-path exec-path
-                                        path-separator)))
+              (setq exec-path (append
+                               (mapcar (lambda (path) (subst-char-in-string ?\\ ?/ path)) paths)
```

Comment:
Ups, just found a possible issue, the original code did also call `directory-file-name` on the paths. Please restore that part of the logic otherwise we may risk issues on certain systems.

Reply:
Upss, restored!